src/Frontend/src/views/ThroughputReportView.vue

```diff
@@ -1,32 +1,68 @@
 <script setup lang="ts">
+import { onMounted, ref } from "vue";
 import routeLinks from "@/router/routeLinks";
 import isRouteSelected from "@/composables/isRouteSelected";
+import { connectionState } from "@/composables/serviceServiceControl";
+import ServiceControlNotAvailable from "@/components/ServiceControlNotAvailable.vue";
+import ThroughputNotAvailable from "@/views/throughputreport/ThroughputNotAvailable.vue";
+import ReportGenerationState from "@/resources/ReportGenerationState";
+import throughputClient from "@/views/throughputreport/throughputClient";
+import { useShowToast } from "@/composables/toast";
+import { TYPE } from "vue-toastification";
+import { useIsThroughputSupported } from "@/views/throughputreport/IsThroughputSupported";
+
+const reportState = ref<ReportGenerationState | null>(null);
+
+onMounted(async () => {
+  reportState.value = await throughputClient.reportAvailable();
+});
+
+async function generateReport() {
+  const spVersion = `spVersion=${encodeURIComponent(window.defaultConfig.version)}`;
+  const fileName = await throughputClient.downloadReport(spVersion);
+
+  if (fileName !== "") {
+    useShowToast(TYPE.INFO, "Report Generated", `Please email ${fileName} to sales@particular.net`);
+  }
+}
 </script>
 
 <template>
-  <div class="container">
-    <div class="row">
-      <div class="col-sm-12">
-        <h1>Throughput Report</h1>
+  <ThroughputNotAvailable />
+  <ServiceControlNotAvailable />
+  <template v-if="connectionState.connected && useIsThroughputSupported()">
```

Comment:
I am not sold on this pattern, it feels like this should use [named slots](https://vuejs.org/guide/components/slots.html#named-slots).
I understand this is replicating today's pattern, but I still think we should change it.
@PhilBastian thoughts?

My take on this is something like:
```html
<ThroughputSupport>
	<slot name="unsupported">
                This probably can default to render <ThroughputNotAvailable />
	</slot>
	<slot>
                .......
	</slot>
</ThroughputSupport>
```

Reply:
slots are for if the decision about what should be displayed needs to be made in the parent. If this fits that criterion then use slots, but currently the decision is being made here, in the child component. If the information is available here then why would you move the decision up to the parent?